pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_demographics_sweep.py

```diff
@@ -0,0 +1,163 @@
+
+        self.file_params["spatial_output"] = True
+        self.file_params["age_output"] = True
+        with patch('pyEpiabm.output._csv_dict_writer.open', mo):
+            # Now test init using valid input
+            dem_sweep = pe.sweep.DemographicsSweep(self.file_params)
+            dem_sweep.bind_population(self.test_population)
+            self.assertEqual(dem_sweep.spatial_output, True)
+            self.assertEqual(dem_sweep.age_output, True)
+            file_name = os.path.join(os.getcwd(),
+                                     self.file_params["output_dir"],
+                                     "demographics.csv")
+            self.assertEqual(dem_sweep.titles, ["id", "age_group",
+                                                "location_x", "location_y",
+                                                "kw_or_chr"])
+            del dem_sweep.writer
+        mo.assert_called_with(file_name, 'w')
+
+    @patch('os.makedirs')
+    def test_write_to_file(self, mock_mkdir):
+
+        if os.path.exists(self.file_params["output_dir"]):
+            os.rmdir(self.file_params["output_dir"])
+
+        mo = mock_open()
+        self.file_params['spatial_output'] = False
+        self.file_params['age_output'] = False
+        with patch('pyEpiabm.output._csv_dict_writer.open', mo):
+            dem_sweep = pe.sweep.DemographicsSweep(self.file_params)
+            dem_sweep.bind_population(self.test_population)
+            person_0_0_0_0_data = {"id": "0.0.0.0", "kw_or_chr": 'K'}
+            person_1_0_0_0_data = {"id": "1.0.0.0", "kw_or_chr": 'C'}
+            person_1_0_0_1_data = {"id": "1.0.0.1", "kw_or_chr": 'X'}
+            with patch.object(dem_sweep.writer, 'write') as mock:
+                dem_sweep()
+                mock.assert_any_call(person_0_0_0_0_data)
+                mock.assert_any_call(person_1_0_0_0_data)
+                mock.assert_any_call(person_1_0_0_1_data)
+                self.assertEqual(mock.call_count, 3)
+        mock_mkdir.assert_called_with(os.path.join(os.getcwd(),
+                                                   self.file_params[
+                                                       "output_dir"]))
+
+        self.file_params['age_output'] = True
+        with patch('pyEpiabm.output._csv_dict_writer.open', mo):
+            dem_sweep = pe.sweep.DemographicsSweep(self.file_params)
+            dem_sweep.bind_population(self.test_population)
+            person_0_0_0_0_data = {"id": "0.0.0.0", "age_group": 5,
+                                   "kw_or_chr": 'K'}
+            person_1_0_0_0_data = {"id": "1.0.0.0", "age_group": 13,
+                                   "kw_or_chr": 'C'}
+            person_1_0_0_1_data = {"id": "1.0.0.1", "age_group": 8,
+                                   "kw_or_chr": 'X'}
+            with patch.object(dem_sweep.writer, 'write') as mock:
+                dem_sweep()
+                mock.assert_any_call(person_0_0_0_0_data)
```

Comment:
This is a very broad assertion. If this is called a fixed number of times (or with a known argument) we should check that instead

Reply:
Is this better?
`                mock.assert_has_calls([call(person_0_0_0_0_data),
                                       call(person_1_0_0_0_data),
                                       call(person_1_0_0_1_data)])
                self.assertEqual(mock.call_count, 3)`

Reply:
That is great!